mlflow/utils/search_utils.py

```diff
@@ -1018,13 +980,5 @@ def _invalid_statement_token_search_model_registry(cls, token):
             return False
         elif token.match(ttype=TokenType.Keyword, values=["AND", "IN"]):
             return False
-        elif token.match(ttype=TokenType.Operator.Comparison, values=["IN"]):
-            # `IN` is a comparison token in sqlparse >= 0.4.0:
-            # https://github.com/andialbrecht/sqlparse/pull/567
-            if token.value == "IN":
-                # This case it represent the IN filter parsed failed.
-                # e.g. "run_id IN"
-                return True
-            return False
```

Comment:
WHy do we need to remove this block?

Reply:
This is code for compatibility handling for sqlparse<4.0.0: 
In sqlparse<4.0.0, IN operator is not a `Comparison` type instance, but a Token type instance with `ttype=TokenType.Operator.Comparison, values=["IN"]`
but, current parsing logic already breaks the logic here under sqlparse<4.0.0 .

Reply:
This code is for compatibility for `sqlparse>=0.4.0` as the comment explains.

Reply:
No. The comment is not clear.
Using `sqlparse>=0.4.0`, this block code will never be run.

Reply:
Test:

```diff
diff --git a/mlflow/utils/search_utils.py b/mlflow/utils/search_utils.py
index 2a3c5dbe6..f69cc5498 100644
--- a/mlflow/utils/search_utils.py
+++ b/mlflow/utils/search_utils.py
@@ -1019,6 +1019,7 @@ class SearchModelUtils(SearchUtils):
         elif token.match(ttype=TokenType.Keyword, values=["AND", "IN"]):
             return False
         elif token.match(ttype=TokenType.Operator.Comparison, values=["IN"]):
+            raise Exception(sqlparse.__version__)
             # `IN` is a comparison token in sqlparse >= 0.4.0:
             # https://github.com/andialbrecht/sqlparse/pull/567
             if token.value == "IN":
```

```
> pytest tests/store/model_registry/test_sqlalchemy_store.py::TestSqlAlchemyStoreSqlite::test_search_model_versions -s

...
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    @classmethod
    def _invalid_statement_token_search_model_registry(cls, token):
        if isinstance(token, (Comparison, Identifier, Parenthesis)):
            return False
        elif token.is_whitespace:
            return False
        elif token.match(ttype=TokenType.Keyword, values=["AND", "IN"]):
            return False
        elif token.match(ttype=TokenType.Operator.Comparison, values=["IN"]):
>           raise Exception(sqlparse.__version__)
E           Exception: 0.4.2


mlflow/utils/search_utils.py:1022: Exception
```

Reply:
You misread above comment, I said `this block code will never be run except invalid filter such as run_id IN`,
Your test is the case `invalid filter such as run_id IN`

Reply:
You're right. I thought the error was thrown in a test case with a valid filter string, but it actually wasn't.